Add generated client with necessary sanitizations

rtweera opened the pull request:
## Purpose
To generate ballerina client code from OpenAPI specifications with necessary sanitations

## Examples

## Checklist
- [ ] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests
- [x] Updated the spec
- [ ] Checked native-image compatibility

rtweera committed:
Add flattened & aligned openapi spec

rtweera committed:
Add sanitations

URL paths were duplicated and long. Add common path to server URL and
remove duplications. Makes the paths easier to read and maintain.

Add examples for parameters in the REST methods to show the correct
format of the parameter values. Helps to avoid validation errors.

rtweera committed:
Fix sanitations.md formatting

rtweera committed:
Generate client, types, utils using openapi tool

rtweera committed:
Merge pull request #1 from rtweera/dev

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/ballerina-platform/module-ballerinax-hubspot.crm.extensions.videoconferencing?pullRequest=3) <br/>All committers have signed the CLA.

Nuvindu commented on docs/spec/openapi.json:
```diff
@@ -0,0 +1,332 @@
+    "cms" : "STARTER"
+  },
+  "x-hubspot-documentation-banner" : "NONE"
+}
```
```suggestion
}

```

Nuvindu commented on docs/spec/sanitations.md:
````diff
@@ -1,24 +1,62 @@
-3. 
+1. Change the `url` property of the objects in servers array
+
+    - **Original**:
+    ```https://api.hubspot.com```
+
+    - **Updated**:
+    ```https://api.hubapi.com/crm/v3/extensions/videoconferencing/settings```
````
```suggestion
    - **Original**:
    `https://api.hubspot.com`

    - **Updated**:
    `https://api.hubapi.com/crm/v3/extensions/videoconferencing/settings`
```

Nuvindu replied:
check for other places

Nuvindu commented on docs/spec/sanitations.md:
````diff
@@ -1,24 +1,62 @@
+
+    - **Original**: No example was present.
+
+    ```json
+        "fetchAccountsUri" : {
+            "type" : "string"
+          }
+    ```
+
+    - **Updated**: Example section added with a sample URI.
+
+    ```json
+        "fetchAccountsUri" : {
+            "type" : "string",
+            "example" : "https://example.xyz/1"
+          }
+    ```
````
```suggestion
    ```json
    {
        "fetchAccountsUri" : {
            "type" : "string"
          }
    }
    ```

    - **Updated**: Example section added with a sample URI.

    ```json
    {
        "fetchAccountsUri" : {
            "type" : "string",
            "example" : "https://example.xyz/1"
    }
    ```
```

Nuvindu commented on docs/spec/sanitations.md:
```diff
@@ -1,24 +1,62 @@
 
-This document records the sanitation done on top of the official OpenAPI specification from HubSpot CRM Video conference. 
+This document records the sanitation done on top of the official OpenAPI specification from HubSpot CRM Video conference.
 The OpenAPI specification is obtained from (TODO: Add source link).
```
Add source link here

Nuvindu commented on docs/spec/sanitations.md:
````diff
@@ -1,24 +1,62 @@
+2. Update the API `paths`
+
+    - **Original**: Paths included the common prefix above `(i.e. /crm/v3/extensions/videoconferencing/settings)` in each endpoint.
+    ```/crm/v3/extensions/videoconferencing/settings/{appId}```
+
+    - **Updated**: Common prefix removed from path endpoints.
+    ```/{appId}```
````
```suggestion
    `/crm/v3/extensions/videoconferencing/settings/{appId}`

    - **Updated**: Common prefix removed from path endpoints.
    `/{appId}`
```

Nuvindu commented on docs/spec/sanitations.md:
````diff
@@ -1,24 +1,62 @@
+
+2. Update the API `paths`
+
+    - **Original**: Paths included the common prefix above `(i.e. /crm/v3/extensions/videoconferencing/settings)` in each endpoint.
+    ```/crm/v3/extensions/videoconferencing/settings/{appId}```
+
+    - **Updated**: Common prefix removed from path endpoints.
````
```suggestion
   - **Original**: Paths included the common prefix above `(i.e. /crm/v3/extensions/videoconferencing/settings)` in each endpoint.
    ```/crm/v3/extensions/videoconferencing/settings/{appId}```

   - **Updated**: Common prefix removed from path endpoints.
```

rtweera replied:
Hi @Nuvindu . Here, I should keep the nesting, but use 3 spaces instead of 4 right?

Nuvindu commented on docs/spec/sanitations.md:
````diff
@@ -1,24 +1,62 @@
+
+3. Add example for fetchAccountsUri object in External settings for `put` method
+
+    - **Original**: No example was present.
+
+    ```json
+        "fetchAccountsUri" : {
+            "type" : "string"
+          }
+    ```
+
+    - **Updated**: Example section added with a sample URI.
````
```suggestion
   - **Original**: No example was present.

    ```json
        "fetchAccountsUri" : {
            "type" : "string"
          }
    ```

   - **Updated**: Example section added with a sample URI.
```

Nuvindu commented on docs/spec/sanitations.md:
````diff
@@ -1,24 +1,62 @@
-3. 
+1. Change the `url` property of the objects in servers array
+
+    - **Original**:
+    ```https://api.hubspot.com```
+
+    - **Updated**:
````
```suggestion
   - **Original**:
    ```https://api.hubspot.com```

   - **Updated**:
```

rtweera committed:
Fix formatting issues for sanitations.md

rtweera committed:
Merge pull request #2 from rtweera/dev

Fix formatting issues for sanitations.md

Nuvindu commented on docs/spec/sanitations.md:
```diff
@@ -1,24 +1,65 @@
+
+   - **Reason**: This simplifies the API paths making them shorter and easier to read.
+
+3. Add example for fetchAccountsUri object in External settings for `put` method
```
```suggestion
3. Add example for `fetchAccountsUri` object in External settings for `put` method
```

rtweera committed:
Update docs/spec/sanitations.md

Co-authored-by: Nuvindu Nirmana <63797478+Nuvindu@users.noreply.github.com>

Nuvindu commented on docs/spec/sanitations.md:
```diff
@@ -1,24 +1,65 @@
-_Updated_: <!-- TODO: Add date --> \
+_Author_:  @rtweera\
+_Created_: 13.02.2025 \
+_Updated_: 14.02.2025 \
```
```suggestion
_Created_: 2025/02/13 \
_Updated_: 2025/02/14 \
```

rtweera committed:
Update docs/spec/sanitations.md

Created and Updated date format fix

Co-authored-by: Nuvindu Nirmana <63797478+Nuvindu@users.noreply.github.com>

rtweera committed:
Add nullable to openapi.json & sanitations.md

Update docs/spec/openapi.json file to add nullable to the fields that
are optional in the API response. This ensures no json parsing errors
when the optional fields are not present in the response.

Update docs/spec/sanitations.md file to reflect the changes done on
docs/spec/openapi.json file.

rtweera committed:
Add description to openapi.json and sanitations.md

Add descrition to the field hapikey in the openapi.json. This makes it
easier for the developers to understand the purpose of the field.

Add necessary changes in the sanitations.md to reflect the changes in
the openapi.json.

rtweera committed:
Regenerate client with necessary sanitations

Generate client, types, and utils with updated sanitations to improve
documentation and better error handling.

rtweera committed:
Merge pull request #3 from rtweera/dev

Updated client
